src/App/App.test.js

```diff
@@ -1,35 +1,215 @@
 import React from 'react'
-import { render, screen, getByPlaceholderText, getByText, waitFor } from '@testing-library/react'
+// import React-DOM from 'react-dom'
+import { render, screen, getByPlaceholderText, getByText, waitFor, getByAltText } from '@testing-library/react'
 import App from './App'
-import API from '../API/API'
-// import Header from '../Header/Header'
 import '@testing-library/jest-dom'
+import API from '../API/API'
+jest.mock('../API/API.js')
 
-describe('App', () => {
-  beforeEach(() => {
-  App.
 
-  render(<App/>)
- })
 
-  it('should start with default login state', () => {
+describe('App', () => {
+
+  it('should start with default state', () => {       
     const defaultApp = new App()
+    expect(defaultApp.state.movies).toStrictEqual([])
+    expect(defaultApp.state.error).toBe('')
+    expect(defaultApp.state.pageView).toBe('Home')
+    expect(defaultApp.state.userRatings).toStrictEqual([])
+    expect(defaultApp.state.singleMovie).toStrictEqual({})
+    expect(defaultApp.state.singleMovieUserRating).toStrictEqual({})
+    expect(defaultApp.state.trailers).toStrictEqual([])
+    expect(defaultApp.state.user).toStrictEqual({id: '', email: '', name: ''})
     expect(defaultApp.state.isLoggedIn).toBe(false)
   })
-
+  
   it('should render header', () => {
-    expect(screen.getByPlaceholderText('Search by title, genre, year')).toBeInTheDocument()
+    const {getByPlaceholderText, getByText} = render(<App/>)
+    expect(getByText('Home')).toBeInTheDocument()
+    expect(getByPlaceholderText('Search by title, genre, year')).toBeInTheDocument()
+    expect(getByText("Your Ratings").className).toBe("hidden")
+    expect(getByText('Logout').className).toBe('hidden')
   })
-
-  it('should hide some header elements in default state', () => {
-    expect(screen.getByText("Your Ratings").className).toBe("hidden");
-    expect(screen.getByText('Logout').className).toBe('hidden')
+  
+  it('should render some header elements if isLoggedIn is true', () => {
+    const { getByText } = render(<App isLoggedIn={true}/>)
```

Comment:
I think this is going to be our main trick for avoiding API and the regular ways data trigger changes on the App and other components